### ISSUE TITLE
Resolution of platform dependencies fails the build with Kotlin 1.9 / Gradle 8

### ISSUE THREAD

**Issue by IgnatBeresnev** [Kotlin/dokka#3153]
## Problem

Due to a combination of reasons, Gradle builds might fail with one of the following problems:

> Task ':redwood-treehouse:dokkaHtmlPartial' uses this output of task ':redwood-protocol-widget:transformIosMainCInteropDependenciesMetadataForIde' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.

or

> Cannot change dependencies of dependency configuration ':vclib-openid:iosMainImplementationDependenciesMetadata' after task dependencies have been resolved

## Cause

It's difficult to pinpoint the exact reason for these errors as multiple factors contribute to it:

* Kotlin 1.9.0 (likely some underlying change to `implementationMetadataConfiguration` in KGP)
* Gradle 8
* Usage of [`cinterop`](https://kotlinlang.org/docs/native-c-interop.html)
* `kotlin.mpp.enableCInteropCommonization` being set to `true`
* Usage of Gradle's `includeBuild`

Depending on the combination, it will lead to one of the mentioned errors.

## Workarounds

Unfortunately, the potential fixes were found late into testing Dokka 1.9.0. We did not want to make rushed decisions that could break unaffected projects, but we also did not want to delay the release even more, so we settled on providing workarounds in Dokka 1.9.0 that are _disabled_ by default.

There are three flags that change the way Dokka resolves platform dependencies. The flags are **mutually exclusive**, please try them in **descending order** (from top to bottom):
* `org.jetbrains.dokka.classpath.useNativeDistributionAccessor=true`
* `org.jetbrains.dokka.classpath.useKonanDistribution=true`
* `org.jetbrains.dokka.classpath.excludePlatformDependencyFiles=true` - last resort, ignores platform dependencies altogether, so you might see `Error class` in signatures.

You can add these flags to your `gradle.properties` or pass them as a project property in CLI:
```bash
./gradlew -Porg.jetbrains.dokka.classpath.useNativeDistributionAccessor=true
```

Please, let us know if any of these workarounds (or especially all of them) didn't help.

## Short-term solution

We aim to ship a well-tested solution that will be enabled by default in Dokka 1.9.10, likely some time in October.

## Long-term solution

We're taking the necessary steps to resolve this problem for good (#3121) and to discover similar compatibility problems very much in advance, before our users experience them. We plan to implement it by Kotlin 2.0 / 2.1, but no later than the first Stable release of Dokka.

**Comment by JakeWharton** [comment]
The first property appears to have worked for Redwood. Thanks! I did not try the others.

**Comment by jeffdgr8** [comment]
The first property resolved the issue in my library as well. Thank you!

By the way, I believe you meant "try them in **_descending_ order**".

**Comment by AngeloFilaseta** [comment]
The first property worked for me too as a workaround. I was struggling very hard before finding this issue :(

**Comment by IgnatBeresnev** [comment]
This problem was fixed in Dokka 1.9.10, so it no longer requires the flags mentioned above.

* If you are using Dokka 1.9.10, you can remove the flags from your configs
* If you are using Dokka < 1.9.0, I suggest you make the jump to 1.9.10, and the problem will be fixed without any additional flags.

Thanks to everyone who tested the solution and provided feedback! I'll close the issue now as we haven't seen any new reports about it recently, so hopefully it's all working as expected

**Comment by gosha212** [comment]
Hi @IgnatBeresnev,
I still can reproduce the issue on my android project after upgrading to gradle 8.3 with dokka 1.9.10.
`./gradlew dokkaDocJar` does work for me, but the publish task fails with the following problem:

A problem was found with the configuration of task ':detox:dokkaJavadoc' (type 'DokkaTask').
  - Gradle detected a problem with the following location: '/Users/georgys/Workplace/Detox/detox-public/detox/android/detox/build/intermediates/compile_r_class_jar/release/R.jar'.
    
    Reason: Task ':detox:dokkaJavadoc' uses this output of task ':detox:generateReleaseRFile' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':detox:generateReleaseRFile' as an input of ':detox:dokkaJavadoc'.
      2. Declare an explicit dependency on ':detox:generateReleaseRFile' from ':detox:dokkaJavadoc' using Task#dependsOn.
      3. Declare an explicit dependency on ':detox:generateReleaseRFile' from ':detox:dokkaJavadoc' using Task#mustRunAfter.
    
    For more information, please refer to https://docs.gradle.org/8.3/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.


Here is the branch: https://github.com/wix/Detox/tree/upgrade/RN73
How to reproduce: 

`cd detox/android`
`./gradlew publish -Dversion=999.999.999 `
